lib/wasix/src/bin_factory/binary_package.rs

```diff
@@ -79,6 +80,36 @@ pub struct BinaryPackage {
 }
 
 impl BinaryPackage {
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn from_dir(
+        dir: &Path,
+        rt: &(dyn Runtime + Send + Sync),
+    ) -> Result<Self, anyhow::Error> {
+        let source = rt.source();
+
+        // since each package must be in its own directory, hash of the `dir` should provide a good enough
+        // unique identifier for the package
+        let hash = sha2::Sha256::digest(dir.display().to_string().as_bytes()).into();
```

Comment:
Minor, but to prevent lossiness from `.display()` we could also use:  `dir.as_os_str().as_encoded_bytes()`.

Reply:
I originally did this, but aren't we on `1.73`? `as_encoded_bytes` was stabilized on `1.74`.